Tools/cases_generator/generate_cases.py

```diff
@@ -234,7 +234,74 @@ def assign(self, dst: StackEffect, src: StackEffect):
     def cast(self, dst: StackEffect, src: StackEffect) -> str:
         return f"({dst.type or 'PyObject *'})" if src.type != dst.type else ""
 
-INSTRUCTION_FLAGS = ['HAS_ARG', 'HAS_CONST', 'HAS_NAME', 'HAS_JUMP']
+
+class InstructionFlags:
+    """Construct and manipulate instruction flags"""
+
+    INSTRUCTION_FLAGS = ['HAS_ARG', 'HAS_CONST', 'HAS_NAME', 'HAS_JUMP']
+    INSTR_FLAG_SUFFIX = "_FLAG"
+
+    def __init__(self, has_arg: bool, has_const: bool, has_name: bool,
+                 has_jump: bool):
+
+        self.data = {
+            'HAS_ARG': has_arg,
+            'HAS_CONST': has_const,
+            'HAS_NAME': has_name,
+            'HAS_JUMP': has_jump,
+        }
```

Comment:
I'd rather see the bits stored as separate attributes. Maybe INSTRUCTION_FLAGS could just be a list of attribute names.